### ISSUE TITLE
Ganeti fails to create instance on Debian wheezy

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#263]
Originally reported of Google Code with ID 207.

```
User JonB on IRC reported a failure on instance creation and burnin using debian 2.4.5 on Debian wheezy. The error was related to the pvs utility output.

Turns out that the command "pvs --noheadings --nosuffix --units=m --unbuffered '--separator=|' -opv_name,vg_name,pv_free,pv_attr" on his machine produced the following broken output:

  /dev/sdb5|dkgnode00|0|a--
  /dev/sda1|xenvg|466568.00|a--kgnod/dev/sda1|xenvg|466568.00|

Downgrading LVM2 from 2.02.88-2 to 2.02.66-5 solved the problem, the output is parseable again by Ganeti and instance creation succeeds.

I was unable to reproduce the bug on a wheezy test machine with the same lvm2 version. I'm opening this bug for future reference.
```
Originally added on 2011-12-08 14:34:53 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Link to Debian bug: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=651425
```
Originally added on 2011-12-08 14:59:14 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Similar issue with Wheezy that I can confirm but it isn't in LVM package versions.  It would be found in parsing the kvm version from 'kvm --help'.  Far as I can tell from reading the code and trying the new regex, it is fixed in 2.5.0rc5.  If there is a 2.4.6 planned I can submit the patch real quick, else I might just build debs for 2.5.0rc.
```
Originally added on 2012-01-22 23:37:38 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Cody, we don't have any plans right now for a 2.4.6, but you could backport the patch to your 2.4 build too.

As to the LVM part, that's weird, it almost looks like a bug in LVM itself. I'll try again with latest wheezy and see what's the status.
```
Originally added on 2012-02-20 10:20:13 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
hi,
still reproducible.

settler1.died.ru: gnt-cluster (ganeti v2.5.0rc5) 2.5.0~rc5
settler2.died.ru: gnt-cluster (ganeti v2.5.0rc5) 2.5.0~rc5

on both nodes:
ii  liblvm2app2.2                        2.02.88-2                      LVM2 application library
ii  lvm2                                 2.02.88-2                      Linux Logical Volume Manager

  PV         VG     Fmt  Attr PSize   PFree    
  /dev/sdb1  xenvg  lvm2 a--  931.51g 901.89g
  /dev/sdc2  vg0    lvm2 a--  232.41g  17.41g

```
Originally added on 2012-03-21 23:56:41 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
using 2.5.0 i can reproduce this on gentoo as well.

i believe this is a lvm bug just based on running the following commands

$ pvs --noheadings --nosuffix --units=m --unbuffered  -opv_name,vg_name,pv_free,pv_attr
  /dev/sdb1  vg1  1905436.00 a-- 
  /dev/sda5  vg0    27740.00 a--v/sdb/dev/sda5  vg0    27740

while removing the pv_attr column results in the following:

$ pvs --noheadings --nosuffix --units=m --unbuffered  -opv_name,vg_name,pv_free
  /dev/sdb1  vg1  1905436.00
  /dev/sda5  vg0    27740.00

```
Originally added on 2012-04-25 04:59:36 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
This is a known issue with LVM version 2.02.88. We need the attr column, so the only fix is to upgrade to a newer LVM version (2.02.96 is known to work well).
```
Originally added on 2012-04-25 07:03:08 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
thanks. i upgraded to 2.0.2.93-r1 on gentoo. not sure if 2.02.96 has been released yet. it fixed the issue from before, however i get different error now between gnt-cluster verify and verify-disks.

$ gnt-cluster verify
Submitted jobs 33, 34
Waiting for job 33 ...
Thu Apr 26 21:08:05 2012 * Verifying cluster config
Thu Apr 26 21:08:05 2012 * Verifying cluster certificate files
Thu Apr 26 21:08:05 2012 * Verifying hypervisor parameters
Thu Apr 26 21:08:05 2012 * Verifying all nodes belong to an existing group
Waiting for job 34 ...
Thu Apr 26 21:08:05 2012 * Verifying group 'default'
Thu Apr 26 21:08:05 2012 * Gathering data (2 nodes)
Thu Apr 26 21:08:06 2012 * Gathering disk information (2 nodes)
Thu Apr 26 21:08:06 2012 * Verifying configuration file consistency
Thu Apr 26 21:08:06 2012 * Verifying node status
Thu Apr 26 21:08:06 2012 * Verifying instance status
Thu Apr 26 21:08:06 2012   - ERROR: instance forge.codebex.lan: volume vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_data missing on node beta.orion.example.lan
Thu Apr 26 21:08:06 2012   - ERROR: instance forge.codebex.lan: volume vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_meta missing on node beta.orion.example.lan
Thu Apr 26 21:08:06 2012   - ERROR: instance forge.codebex.lan: volume vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_data missing on node alpha.orion.example.lan
Thu Apr 26 21:08:06 2012   - ERROR: instance forge.codebex.lan: volume vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_meta missing on node alpha.orion.example.lan
Thu Apr 26 21:08:06 2012 * Verifying orphan volumes
Thu Apr 26 21:08:06 2012 * Verifying N+1 Memory redundancy
Thu Apr 26 21:08:06 2012 * Other Notes
Thu Apr 26 21:08:06 2012 * Hooks Results

while gnt-cluster verify-disks returns the following

$ gnt-cluster verify-disks
Submitted jobs 36
Waiting for job 36 ...

every disk i create ends up with the above error.
```
Originally added on 2012-04-27 01:12:20 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
thanks. i upgraded to 2.0.2.93-r1 on gentoo. not sure if 2.02.96 has been released yet. it fixed the issue from before, however i get different error now between gnt-cluster verify and verify-disks.

$ gnt-cluster verify
Submitted jobs 33, 34
Waiting for job 33 ...
Thu Apr 26 21:08:05 2012 * Verifying cluster config
Thu Apr 26 21:08:05 2012 * Verifying cluster certificate files
Thu Apr 26 21:08:05 2012 * Verifying hypervisor parameters
Thu Apr 26 21:08:05 2012 * Verifying all nodes belong to an existing group
Waiting for job 34 ...
Thu Apr 26 21:08:05 2012 * Verifying group 'default'
Thu Apr 26 21:08:05 2012 * Gathering data (2 nodes)
Thu Apr 26 21:08:06 2012 * Gathering disk information (2 nodes)
Thu Apr 26 21:08:06 2012 * Verifying configuration file consistency
Thu Apr 26 21:08:06 2012 * Verifying node status
Thu Apr 26 21:08:06 2012 * Verifying instance status
Thu Apr 26 21:08:06 2012   - ERROR: instance forge.example.lan: volume vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_data missing on node beta.orion.example.lan
Thu Apr 26 21:08:06 2012   - ERROR: instance forge.example.lan: volume vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_meta missing on node beta.orion.example.lan
Thu Apr 26 21:08:06 2012   - ERROR: instance forge.example.lan: volume vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_data missing on node alpha.orion.example.lan
Thu Apr 26 21:08:06 2012   - ERROR: instance forge.example.lan: volume vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_meta missing on node alpha.orion.example.lan
Thu Apr 26 21:08:06 2012 * Verifying orphan volumes
Thu Apr 26 21:08:06 2012 * Verifying N+1 Memory redundancy
Thu Apr 26 21:08:06 2012 * Other Notes
Thu Apr 26 21:08:06 2012 * Hooks Results

while gnt-cluster verify-disks returns the following

$ gnt-cluster verify-disks
Submitted jobs 36
Waiting for job 36 ...

every disk i create ends up with the above error.
```
Originally added on 2012-04-27 01:14:47 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
just to double check things, i tried the latest version of lvm2 2.02.95-r1 and i get the same results as the 2.02.93-r1.
```
Originally added on 2012-04-27 02:04:01 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
trying to provide as much information as possible. after i tried installing a linux distro in the instance, verify-disks starts to report errors.

Submitted jobs 69
Waiting for job 69 ...
Instance forge.example.lan has missing logical volumes:
	alpha.orion.example.lan /dev/vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_data
	alpha.orion.example.lan /dev/vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_meta
	beta.orion.example.lan /dev/vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_data
	beta.orion.example.lan /dev/vg1/44490f5b-2bd5-43fa-b8db-17ca1232f254.disk0_meta
You need to replace or recreate disks for all the above instances if this message persists after fixing broken nodes.

```
Originally added on 2012-04-27 02:36:08 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
so debugging the issue, it fails on the regex for lvs. i've included a patch that works with lvm2 versions 2.02.93-r1 and up on gentoo. just need to check for 8 characters for the lvs attribute instead of 6. hope this helps others out there.
```
Originally added on 2012-04-27 21:41:21 +0000 UTC.

Attachments: 
[lvsregex.patch](https://storage.googleapis.com/google-code-attachments/ganeti/issue-207/comment-11/lvsregex.patch)

**Comment by Ganeti-Issues-Migrator** [comment]
```
Sorry I didn't watch this bug closely. This is a known issue and was fixed in commit 0304f0ec, which was included in the 2.5.0rc6 release.

We don't have plans for a new 2.4 release, now that 2.5.0 has been released.
```
Originally added on 2012-04-27 21:53:28 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.5

**Comment by Ganeti-Issues-Migrator** [comment]
```
i was using the latest version of ganeti (2.5.0) with the following sha1sum fd049c7bbdc73268b8f8940627a1e7299e7a55f4. Can you confirm the that the issue is fixed in the final release?
```
Originally added on 2012-04-30 14:23:53 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Ah. We fixed _another_ issue with lvs output, but not the regex. Yes, that will have to fixed too.

Thanks for the clarification, I misunderstood things.
```
Originally added on 2012-04-30 20:24:42 +0000 UTC.

Changed State: Accepted

**Comment by Ganeti-Issues-Migrator** [comment]
```
This was released as part of 2.5.1; I'll mark it closed.
```
Originally added on 2012-05-22 15:38:28 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
Migrate old issues from 'fixed' to 'released'.
```
Originally added on 2012-10-22 09:16:17 +0000 UTC.

Changed State: Released
